### PR TITLE
Fix issues in registration form:

### PR DESCRIPTION
**Pull request by javeriaejaz-arbisoft**
- Add foreign key support in Registration Form
- Update migrations in applications and registration_extension

### CODE REVIEW

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/541/workflows/workflows_id/jobs/541/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/admin.py (100%)
openedx/adg/lms/applications/models.py (66.7%): Missing lines 54
openedx/adg/lms/registration_extension/forms.py (100%)
openedx/adg/lms/registration_extension/models.py (100%)
openedx/adg/lms/user_authn_override/login_form.py (100%)
-------------
Total:   13 lines
Missing: 1 line
Coverage: 92%
-------------
```

File: openedx/adg/lms/applications/admin.py [hunk 1]
```diff
@@ -23,9 +23,8 @@ class UserApplicationAdmin(admin.ModelAdmin):
     """
     Django admin class for UserApplication
     """
-    fields = ('user', 'city', 'organization', 'linkedin_url', 'resume', 'cover_letter_file', 'cover_letter',)
-    list_display = ('id', 'user_email', 'city',)
-    list_filter = ('city', 'user__profile__country')
+    list_display = ('id', 'user_email', 'city', 'business_line',)
+    list_filter = ('business_line', 'city', 'user__profile__country')
```
Comment (asadali145) [inline openedx/adg/lms/applications/admin.py:27]: Can we remove country as this is of no use for now.

**Review by ahmed-faraz46 — APPROVED** [review]
PR was already reviewed and merged, we just change business_line to FK, but to to commit amend, it is showing 8 files changed. So LGTM 👍

**Review by asadali145 — APPROVED** [review]
Just remove the country, Apart from that LGTM👍

File: openedx/adg/lms/registration_extension/models.py [hunk 2]
```diff
@@ -14,12 +14,13 @@ class ExtendedUserProfile(models.Model):
     """
 
     user = models.OneToOneField(
-        User, unique=True, db_index=True, related_name='extended_profile', on_delete=models.CASCADE)
+        User, db_index=True, related_name='extended_profile', on_delete=models.CASCADE)
 
-    birth_date = models.DateField(verbose_name=_('Birth Date'), null=True)
-    saudi_national = models.BooleanField(verbose_name=_('Saudi National'), null=True)
+    birth_date = models.DateField(verbose_name=_('Birth Date'), null=True, blank=True)
+    saudi_national = models.BooleanField(verbose_name=_('Saudi National'), null=True, blank=True)
```
Comment (zahid-arbisoft) [inline openedx/adg/lms/registration_extension/models.py:20]: Shouldn't it be `blank=False`?

File: openedx/adg/lms/registration_extension/models.py [hunk 3]
```diff
@@ -14,12 +14,13 @@ class ExtendedUserProfile(models.Model):
...
 
-    company = models.OneToOneField(BusinessLine, on_delete=models.CASCADE, null=True)
+    company = models.ForeignKey(BusinessLine, verbose_name=_('Company'),
+                                on_delete=models.CASCADE, null=True, blank=True)
```
Comment (zahid-arbisoft) [inline openedx/adg/lms/registration_extension/models.py:23]: Same, BusinessLine is also required and `blank=True` is not required here

File: openedx/adg/lms/registration_extension/models.py (hunk 2 above)
Comment (javeriaejaz-arbisoft) [reply openedx/adg/lms/registration_extension/models.py:20]: No, Registration Form is also using this model. they are field are optional as they are not required in Registration Form.

File: openedx/adg/lms/registration_extension/models.py (hunk 3 above)
Comment (javeriaejaz-arbisoft) [reply openedx/adg/lms/registration_extension/models.py:23]: Same reason as above

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/542/workflows/workflows_id/jobs/542/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/admin.py (100%)
openedx/adg/lms/applications/models.py (66.7%): Missing lines 54
openedx/adg/lms/registration_extension/forms.py (100%)
openedx/adg/lms/registration_extension/models.py (100%)
openedx/adg/lms/user_authn_override/login_form.py (100%)
-------------
Total:   13 lines
Missing: 1 line
Coverage: 92%
-------------
```

**Review by zahid-arbisoft — APPROVED** [review]

**Pull request merged**